src/basic/calendarspec.c

```diff
@@ -876,6 +884,53 @@ static int parse_calendar_time(const char **p, CalendarSpec *c) {
         return r;
 }
 
+static int is_valid_timezone(char *tz, bool *result)
+{
+        pid_t pid;
+
+        if (tz == NULL || strlen(tz) == 0) {
+                *result = false;
+                return 0;
+        }
+
+        bool *shared = mmap(NULL, sizeof *shared, PROT_READ|PROT_WRITE, MAP_SHARED|MAP_ANONYMOUS, -1, 0);
```

Comment:
nitpick: see CODING_STYLE, we avoid defining variables in the middle of functions, and we avoid mixing variable declarations and function invocations in one line